packages/core/src/provider/initConfigs.ts

```diff
@@ -40,28 +40,39 @@ export type InitConfigsOptions = {
 
 const normalizeEnvironmentsConfigs = (
   normalizedConfig: NormalizedConfig,
+  rootPath: string,
 ): Record<string, NormalizedEnvironmentConfig> => {
   const { environments, dev, server, provider, ...rsbuildSharedConfig } =
     normalizedConfig;
   if (environments && Object.keys(environments).length) {
     return Object.fromEntries(
-      Object.entries(environments).map(([name, config]) => [
-        name,
-        {
+      Object.entries(environments).map(([name, config]) => {
+        const environmentConfig = {
           ...mergeRsbuildConfig(
             rsbuildSharedConfig,
             config as unknown as NormalizedEnvironmentConfig,
           ),
           dev,
           server,
-        },
-      ]),
+        };
+
+        if (!environmentConfig.source.entry) {
+          // @ts-expect-error
```

Comment:
why @ts-expect-error here

Reply:
Complex type issues, environmentConfig extends normalizedConfig, and it define as readonly.